clients/spark/core/src/main/scala/io/treeverse/clients/GCBackupAndRestore.scala

```diff
@@ -0,0 +1,180 @@
+package io.treeverse.clients
+
+import org.apache.hadoop.conf.Configuration
+
+import scala.util.control.Breaks._
+import org.apache.hadoop.tools.DistCp
+import org.apache.hadoop.fs.{FileSystem, Path}
+import org.apache.spark.sql.{DataFrame, SparkSession}
+
+import java.net.URI
+
+class GCBackupAndRestore {}
+
+object GCBackupAndRestore {
+  val S3AccessKeyName = "fs.s3a.access.key"
+  val S3SecretKeyName = "fs.s3a.secret.key"
+  val AzureStorageAccountKeyName = "fs.azure.account.key"
+  lazy val spark = SparkSession.builder().appName("GCBackupAndRestore").getOrCreate()
+
+  /** Validates that args are the following:
+   *  1. address of parquet that includes the relative paths of files to backup or restore, created by a gc run
+   *     2. src: the namespace to backup or restore objects from/to (i.e. repo storage namespace, or an external location compatibly)
+   *     3. backup/restore destination: the namespace to backup or restore objects from/to (i.e. an external location or repo storage namespace compatibly)
+   *     4. Object storage type: "s3" or "azure"
+   */
+  def validateArgs(args: Array[String]): Unit = {
+    if (args.length != 4) {
+      Console.err.println(
+        "Usage: ... <objects list location> <src namespace> <dst namespace> <storage type>"
+      )
+      System.exit(1)
+    }
+
+    val objectsListPath = args(0)
+    if (!objectsListPath.endsWith("parquet")) {
+      Console.err.println(
+        "<objects list path> must be of type parquet"
+      )
+      System.exit(1)
+    }
```

Comment:
Is this the only file suffix ever used for Parquet?  And with no case changes?  Otherwise it's a needless limitation on the command.

If the goal is only to accept files from one run_id then let's specify just the run_id.  Otherwise let's not force the type from the name.